library/ssl_tls.c

```diff
@@ -8440,6 +8444,35 @@ static int ssl_handshake_init( mbedtls_ssl_context *ssl )
     {
         ssl->handshake = mbedtls_calloc( 1, sizeof(mbedtls_ssl_handshake_params) );
     }
+#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
+    /* If the buffers are too small - reallocate */
+    if( ssl->in_buf_len < MBEDTLS_SSL_IN_BUFFER_LEN )
+    {
+        resized_buffer = mbedtls_calloc( 1, MBEDTLS_SSL_IN_BUFFER_LEN );
+        memcpy( resized_buffer, ssl->in_buf, ssl->in_buf_len );
+        mbedtls_platform_zeroize( ssl->in_buf, ssl->in_buf_len );
+        mbedtls_free( ssl->in_buf );
```

Comment:
What's the impact on peak heap usage with this approach? Is there a way we can avoid increasing the peak heap usage and still meet our other requirements?